content/docs/atom-guide/dev-guide/shaders/azsl/_index.md

```diff
@@ -1,20 +1,468 @@
+  
+In general, structs are supported just like in HLSL. struct has no constructors/destructors and as such, do not accept default member initializers.  
+  
+Just like in HLSL, struct has the particularity to be data only (no methods).
+
+### Classes
+Classes are supported in AZSL with the exact same expectations of how they are supported in HLSL. They can only be declared/defined in global scopes (outside of SRGs), but an SRG can instantiate class type variables.  
+  
+Like in HLSL, classes can inherit from interfaces to be forced to respect a minimal method set.  
+  
+Unlike in HLSL, AZSL forbids class inheritance among classes.  
+  
+Classes has no constructors/destructors and as such, do not accept default member initializers.
+
+### Matrices
+AZSL follows the HLSL convention to use column major matrices by default.  
+  
+However, in the O3DE Runtime, the default convention is row major matrices.  
+  
+When declaring a matrix, the order can be specified using the row_major or column_major keywords.  
+  
+    ShaderResourceGroupSemantic MatrixExample
+    {
+        FrequencyId = 1;
+    };
+      
+    ShaderResourceGroup MatrixExampleSRG : MatrixExample
+    {
+                     float2x3 defaultMatrix;
+        row_major    float2x3 rowMajorMatrix;
+        column_major float2x3 colMajorMatrix;
+    };
+  
+### Arrays And Unbounded Arrays
+From the point of view of defining arrays, the rules and limitations are exactly the same as HLSL. Examples:  
+  
+    ShaderResourceGroup PerObject : BindingPerObject
+    {
+        float m_arrayOfFloats[64]; // GOOD. Arrays of fundamental types must be of constant size.
+     
+        struct SomeStruct
+        {
+            int m_integer;
+            bool m_boolean;
+        }
+        static const int MyArraySize = 32;
+        SomeStruct m_arrayOfStructs[MyArraySize]; // GOOD. Arrays of packable types (structs & fundamentals) require constant size definition.
+     
+        int m_unboundedArrayOfIntegers[]; // ERROR. Unbounded arrays are not supported for fundamental(packables, in general) data types.
+     
+        Texture2D<float4> m_textures[MyArraySize]; // GOOD. Contant size array of Resource View types is ok.
+         
+        ConstantBuffer<SomeStruct> m_unboundedArrayOfStructs[]; // GOOD. Unbounded array of Resource View type is ok. BUT, there are some restrictions.
+    }
+  
+To learn more about what are the limitations when declaring unbounded arrays, go here: [Binding Rules For Unbounded Arrays.](unbounded-arrays)  
+  
+### <a name="partial"></a> Partial ShaderResourceGroup Definitions.
+Before talking about `partial ShaderResourceGroup`, it's important to talk about [ShaderResourceGroupSemantic](srg-semantics), which is always required when declaring a `ShaderResourceGroup`.  
+Go to this section to read about [ShaderResourceGroupSemantic](srg-semantics).  
```

Comment:
- Only need one link to the docs. 
- Switch to "partial SRGs" when introducing the concept. Later, use `partial ShaderResourceGroup` in code style when talking about defining them in code. 
```suggestion
Before talking about partial SRGs, it's important to talk about SRG Semantics, which is always required when declaring a `ShaderResourceGroup`. Read about SRG Semantics in [Shader Resource Group Semantics](srg-semantics).
```

Reply:
Fixed.